### PR TITLE
added iToL bar chart function

### PR DESCRIPTION
**Pull request by qiyunzhu**
@anupriyatripathi I have implemented the very function following our discussion last year. This function takes the sample BIOM table and the sample metadata, and a user-defined categorical metadata column, and generate a iToL-readable Q2 artifact which will render a multi-bar chart.

Example command:

```bash
python q2_qemistree/_itol_metadata.py \
--classified-feature-data classified-feature-data.qza \
--classyfire-level subclass \
--color-file-path itol_colors.txt \
--label-file-path itol_labels.txt \
--sample-feature-table qe_qert_qiime2_table.qza \
--sample-metadata qe_qert_qiime2_metadata.tsv \
--sample-metadata-column ATTRIBUTE_Sample_type2 \
--barchart-file-path itol_bars.qza
```

In this case, the metadata column is `ATTRIBUTE_Sample_type2`. It has five categories. The output file is `itol_bars.qza`. This file can be directly dragged & dropped into the iToL interface.

The visual effect of the result is like (based on a subsample of 100 taxa):
output

### CODE REVIEW

**Comment by qiyunzhu** [comment]
Hi @anupriyatripathi Thanks for your comment! Will get to it.

File: q2_qemistree/_itol_metadata.py [hunk 1]
```diff
@@ -11,15 +11,16 @@
...
 import seaborn as sns
 import click
-from qiime2 import Artifact
+import biom
+from qiime2 import Artifact, Metadata
 
 
 def classyfire_to_colors(classified_feature_data: pd.DataFrame,
-                         feature_data_column: str, color_palette: str):
+                         classyfire_level: str, color_palette: str):
```
Comment (anupriyatripathi) [inline q2_qemistree/_itol_metadata.py]: Can we use `feature_data_column` instead of `classyfire_level`

Comment (qiyunzhu) [reply q2_qemistree/_itol_metadata.py]: @anupriyatripathi done!

File: q2_qemistree/_itol_metadata.py [hunk 2]
```diff
@@ -72,18 +83,45 @@ def get_itol_visualization(classified_feature_data: str,
...
-                ms2_compound = fdata.loc[idx, 'ms2_compound']
-                if pd.notna(ms2_compound) and not ms2_compound.isspace():
-                    label = ms2_compound
-                else:
-                    label = fdata.loc[idx, feature_data_column]
-                fh.write(idx + '\t' + label + '\n')
-        else:
-            for idx in fdata.index:
-                label = fdata.loc[idx, feature_data_column]
-                fh.write(idx + '\t' + label + '\n')
```
Comment (anupriyatripathi) [inline q2_qemistree/_itol_metadata.py]: @qiyunzhu We are missing this chunk of code from your PR as well? Is there a chance you need to update your branch to the current master and then update your PR?

File: q2_qemistree/_itol_metadata.py [hunk 3]
```diff
@@ -45,11 +46,23 @@ def classyfire_to_colors(classified_feature_data: pd.DataFrame,
...
                            color_file_path: str = './itol_colors.txt',
                            label_file_path: str = './itol_labels.txt',
-                           color_palette: str = 'husl'):
+                           color_palette: str = 'husl',
+                           sample_feature_table: str = None,
+                           sample_metadata: str = None,
+                           sample_metadata_column: str = None,
+                           barchart_file_path: str = './itol_bars.qza'):
     '''This function creates iTOL metadata files to specify clade colors and
     tip labels based on Classyfire annotations.'''
```
[the remark this answers is not in the document]
Comment (anupriyatripathi) [reply q2_qemistree/_itol_metadata.py]: @qiyunzhu can we add a line about this new functionality?

**Comment by qiyunzhu** [comment]
@anupriyatripathi Hope it works this time!

**Comment by qiyunzhu** [comment]
@anupriyatripathi Thanks! What kind of docstring is expected here? Currently there is a one-line docstring, consistent with other functions in the same script.

File: q2_qemistree/_itol_metadata.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anupriyatripathi) [reply q2_qemistree/_itol_metadata.py]: @qiyunzhu Could you update the doc-string to something like below:
"This function creates iTOL metadata files to specify clade colors and tip labels based on Classyfire annotations. It also adds bar plots of the abundance of features stratified by user-specified sample metadata column."

It should be ready to merge after this. Thanks!

[the remark this answers is not in the document]
Comment (qiyunzhu) [reply q2_qemistree/_itol_metadata.py]: Got it! Done! @anupriyatripathi

**Pull request merged**